### PR TITLE
start moving things to ets

### PR DESCRIPTION
**Pull request by StefanYohansson**

### CODE REVIEW

File: lib/redix/pubsub/connection.ex [hunk 1]
```diff
@@ -213,93 +189,34 @@ defmodule Redix.PubSub.Connection do
...
 
   defp register_unsubscription(%{subscriptions: subscriptions} = state, kind, targets, subscriber) do
-    msg_kind =
-      case kind do
-        :unsubscribe -> :unsubscribed
-        :punsubscribe -> :punsubscribed
+    Enum.map(targets, fn channel ->
+      case :ets.lookup(subscriptions, channel) do
+        [] -> nil
+        ls ->
```
Comment (diasbruno) [inline lib/redix/pubsub/connection.ex]: ls -> `[{channel, pids}]`

File: lib/redix/pubsub/connection.ex [hunk 2]
```diff
@@ -213,93 +189,34 @@ defmodule Redix.PubSub.Connection do
...
+          Enum.map(ls, fn ({ref, _pids}) -> Process.demonitor(ref))
```
Comment (diasbruno) [inline lib/redix/pubsub/connection.ex]: ls -> `pids`

File: lib/redix/pubsub/connection.ex [hunk 3]
```diff
@@ -310,9 +227,9 @@ defmodule Redix.PubSub.Connection do
   defp handle_pubsub_msg(%{subscriptions: subscriptions} = state, ["message", channel, payload]) do
     message = message(:message, %{channel: channel, payload: payload})
 
-    case Map.fetch(subscriptions, {:channel, channel}) do
-      {:ok, users} -> Enum.each(users, fn {subscriber, _monitor} -> send(subscriber, message) end)
-      :error -> Logger.warn "{:channel, #{channel}} was not subscribed."
+    case :ets.lookup(subscriptions, channel) do
+      [] -> Logger.warn "{:channel, #{channel}} was not subscribed."
+      ls -> Enum.each(ls, fn {_ref, pid} -> send(pid, message) end)
```
Comment (diasbruno) [inline lib/redix/pubsub/connection.ex]: ls -> `[{channel, pids}]`

File: lib/redix/pubsub/connection.ex [hunk 4]
```diff
@@ -326,9 +243,10 @@ defmodule Redix.PubSub.Connection do
        ]) do
     message = message(:pmessage, %{channel: channel, pattern: pattern, payload: payload})
 
-    case Map.fetch(subscriptions, {:channel, channel}) do
-      {:ok, users} -> Enum.each(users, fn {subscriber, _monitor} -> send(subscriber, message) end)
-      :error -> Logger.warn "{:channel, #{channel}} was not subscribed."
+    case :ets.lookup(subscriptions, channel) do
+      [] -> Logger.warn "{:channel, #{channel}} was not subscribed."
+      ls -> Enum.each(ls, fn {_ref, pid} -> send(pid, message) end)
```
Comment (diasbruno) [inline lib/redix/pubsub/connection.ex]: ls -> `[{channel, pids}]`

File: lib/redix/pubsub/connection.ex (hunk 1 above)
Comment (StefanYohansson) [reply lib/redix/pubsub/connection.ex]: done

File: lib/redix/pubsub/connection.ex (hunk 2 above)
Comment (StefanYohansson) [reply lib/redix/pubsub/connection.ex]: done

File: lib/redix/pubsub/connection.ex (hunk 3 above)
Comment (StefanYohansson) [reply lib/redix/pubsub/connection.ex]: done

File: lib/redix/pubsub/connection.ex (hunk 4 above)
Comment (StefanYohansson) [reply lib/redix/pubsub/connection.ex]: done

**Comment by StefanYohansson** [comment]
```
Redix.PubSubTest
  * test disconnections/reconnections (521.8ms)
  * test recipients are monitored and the connection unsubcribes when they go down (106.7ms)
  * test subscribing the same pid to the same channel more than once has no effect (503.6ms)
  * test :exit_on_disconnection option (25.4ms)
  * test subscribe/unsubscribe flow (502.2ms)
  * test pubsub: unsubscribing a recipient doesn't affect other recipients (1004.9ms)
  * test after unsubscribing from a channel, resubscribing one recipient resubscribes correctly (2.7ms)


Finished in 2.9 seconds
7 tests, 0 failures
```

**Pull request merged**